extensions/database/module/scripts/index/database-source-ui.js

```diff
@@ -241,12 +241,15 @@ Refine.DatabaseSourceUI.prototype._editConnection = function(connectionInfo) {
         success: function(settings) {
              if(settings){
               $( "#menuListUl" ).empty();
+              var menuList = $('#menuListUl');
               var items = [];
               $.each(settings.savedConnections,function(index,savedConnection){
 
-                 items.push('<li class="pure-menu-item sc-list"><a href="#" class="pure-menu-link context-menu-one">'
-                    + '<span class="context-menu-text" >' + savedConnection.connectionName + '</span>'
-                    + '<span class="sc-context-more-vert pull-right"> </span></a></li>');
+                 var li = $('<li class="pure-menu-item sc-list"></li>').appendTo(menuList);
+                 var a = $('<a href="#" class="pure-menu-link context-menu-one"></a>').appendTo(li);
+                 $('<span class="context-menu-text"></span>').text(savedConnection.connectionName)
+                      .appendTo(a);
+                 $('<span class="sc-context-more-vert pull-right"> </span>').appendTo(a);
```

Comment:
This is a better way to construct DOM elements in jQuery: just use the `text()` method of a jQuery element, which already does the escaping for you.
In general, all user-supplied text should be inserted in this way in the UI. Manually concatenating HTML code in JavaScript is a bad sign.